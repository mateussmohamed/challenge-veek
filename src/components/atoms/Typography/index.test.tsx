import React from 'react'

import { Typography } from './index'
import withThemeProvider from '../../theme/withThemeProvider'

describe('[<Typography />]', () => {
  const wrap = props => withThemeProvider(() => <Typography {...props} />)

  test('matches to snapshot', () => {
    const wrapper = wrap({ variant: 'body' })

    expect(wrapper).toMatchSnapshot()
  })

  describe('when pass variant prop', () => {
    test('should equals the title', () => {
      const wrapper = wrap({ children: 'title', variant: 'title' })

      expect(wrapper).toMatchSnapshot()
    })

    test('should equals the title2', () => {
      const wrapper = wrap({ children: 'title2', variant: 'title2' })

      expect(wrapper).toMatchSnapshot()
    })

    test('should equals the subtitle', () => {
      const wrapper = wrap({ children: 'subtitle', variant: 'subtitle' })

      expect(wrapper).toMatchSnapshot()
    })

    test('should equals the subtitle2', () => {
      const wrapper = wrap({ children: 'subtitle2', variant: 'subtitle2' })

      expect(wrapper).toMatchSnapshot()
    })

    test('should equals the body', () => {
      const wrapper = wrap({ children: 'body', variant: 'body' })

      expect(wrapper).toMatchSnapshot()
    })

    test('should equals the legend', () => {
      const wrapper = wrap({ children: 'legend', variant: 'legend' })

      expect(wrapper).toMatchSnapshot()
    })
  })
})
