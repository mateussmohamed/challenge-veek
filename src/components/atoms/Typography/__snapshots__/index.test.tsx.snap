// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[<Typography />] matches to snapshot 1`] = `[Function]`;

exports[`[<Typography />] when pass variant prop should equals the body 1`] = `[Function]`;

exports[`[<Typography />] when pass variant prop should equals the legend 1`] = `[Function]`;

exports[`[<Typography />] when pass variant prop should equals the subtitle 1`] = `[Function]`;

exports[`[<Typography />] when pass variant prop should equals the subtitle2 1`] = `[Function]`;

exports[`[<Typography />] when pass variant prop should equals the title 1`] = `[Function]`;

exports[`[<Typography />] when pass variant prop should equals the title2 1`] = `[Function]`;
